design-proposals/vm-delete-protection.md

```diff
@@ -0,0 +1,171 @@
+# Overview
+
+VirtualMachine Delete Protection prevents the accidental deletion of VMs from a cluster. As long as deletion protection
+is activated for certain VM objects, they cannot be deleted unless the protection is intentionally removed.
+
+## Motivation
+
+KubeVirt clusters are often managed by automations, CLI commands, 3rd party tools or UI actions, for example to deploy,
+reconfigure or delete VMs. These automations may result in accidental deletion of VMs that should have not been deleted.
+Affected VMs could be business critical to application/s or services the cluster is running, this could cause them a
+degradation or out of service, e.g., a SQL server is not available to handle requests.
+Moreover, the affected VMs may contain crucial data if underlining PVC is deleted as a result of a cascaded delete. 
+Therefore, the VM deletion protection aims to avoid accidental degradation of applications or loss of data.
+
+## Goals
+
+- Create a delete protection for VirtualMachine objects.
+
+## Non Goals
+
+- Create a delete protection for VirtualMachineInstance objects.
+- Create a delete protection for virt-launcher Pod objects.
+
+## Definition of Users
+
+VM owners with permissions to delete VMs.
+Cluster administrators with permissions to delete VMs.
+
+## User Stories
+
+As a user who has the permissions to delete VMs, I want to protect my VMs from accidental deletion.
```

Comment:
I'd simplify this and just say `As a VM Owner,` the fact the owner has permissions to delete the VM isn't relevant to the fact that they want to protect it from accidental deletion by them or a third party.

Reply:
Done.